compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BFiniteType.java

```
@@ -65,6 +65,41 @@ public String toString() {
         return joiner.toString();
     }
 
+    @Override
+    public boolean hasImplicitInitialValue() {
+        // Has NIL element as a member.
+        if (valueSpace.stream().anyMatch(value -> value.type.tag == TypeTags.NIL)) {
+            return true;
+        }
+        // All of the element are from same type. And elements contains implicit initial value.
+        if (!valueSpace.isEmpty()) {
```

Can we invert the condition and fail fast? 
```java
if (valueSpace.isEmpty()) { 
   return false;
}
```